src/crypto/oaes_lib.c

```diff
@@ -1272,8 +1272,10 @@ OAES_RET oaes_encrypt( OAES_CTX * ctx,
 		memcpy( _block, c + 2 * OAES_BLOCK_SIZE + _i, _block_size );
 		
 		// insert pad
-		for( _j = 0; _j < OAES_BLOCK_SIZE - _block_size; _j++ )
-			_block[ _block_size + _j ] = _j + 1;
+		for( _j = 0; _j < OAES_BLOCK_SIZE - _block_size; _j++ ) {
+			uint8_t offset = _block_size + _j;
+			_block[ offset ] = _j + 1;
+		}
```

Comment:
Does the warning go away if you change the condition in the original for loop to:

`_block_size + _j < OAES_BLOCK_SIZE`

(Which is arguably clearer code anyway).

Reply:
Sadly not:

```
/home/ianmacd/src/oxen-core/src/crypto/oaes_lib.c: In function ‘oaes_encrypt’:
/home/ianmacd/src/oxen-core/src/crypto/oaes_lib.c:1276:52: error: writing 1 byte into a region of size 0 [-Werror=stringop-overflow=]
 1276 |                         _block[ _block_size + _j ] = _j + 1;
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
/home/ianmacd/src/oxen-core/src/crypto/oaes_lib.c:1269:25: note: at offset 16 into destination object ‘_block’ of size 16
 1269 |                 uint8_t _block[OAES_BLOCK_SIZE];
      |                         ^~~~~~
cc1: all warnings being treated as errors
make[2]: *** [src/crypto/CMakeFiles/cncrypto.dir/build.make:286: src/crypto/CMakeFiles/cncrypto.dir/oaes_lib.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:1475: src/crypto/CMakeFiles/cncrypto.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

Reply:
> The problem is with line 1276, not 1275.

I know, but for some reason gcc isn't noticing that that variable can't be too high to begin with, and was hoping rearranging the condition would make it notice.  But sadly not.